Tests/UnitTests/Caching/PostedTransactionCacheTests.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PostedTransactionCacheTests.swift
+//
+//  Created by Nacho Soto on 7/28/23.
+
+import Foundation
+import Nimble
+import XCTest
+
+@testable import RevenueCat
+
+class PostedTransactionCacheTests: TestCase {
+
+    private var userDefaults: UserDefaults!
+    private var deviceCache: DeviceCache!
+    private var cache: PostedTransactionCache!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        self.userDefaults = try XCTUnwrap(.init(suiteName: UUID().uuidString))
+        self.deviceCache = .init(sandboxEnvironmentDetector: MockSandboxEnvironmentDetector(isSandbox: true),
+                                 userDefaults: self.userDefaults)
+        self.cache = .init(deviceCache: self.deviceCache)
+    }
+
+    func testNoPostedTransactions() {
+        expect(self.cache.hasPostedTransaction(MockStoreTransaction())) == false
+    }
+
+    func testSavesFirstTransaction() {
+        let transaction = MockStoreTransaction()
+
+        self.cache.savePostedTransaction(transaction)
+        expect(self.cache.hasPostedTransaction(transaction)) == true
+    }
+
+    func testSaveMultipleTransactions() {
```

Comment:
we should add a test to verify the behavior if the same transaction gets posted over and over (like if it's not finished)

Reply:
basically to ensure that we don't grow the cache forever

Reply:
It's stored as a `Set`, but yeah I can add that.

Reply:
Done:
```swift
func testHandlePurchasedTransactionMultipleTimesPostsItOnce() throws {
    let transactionData = PurchasedTransactionData(
        appUserID: "user",
        source: .init(isRestore: false, initiationSource: .queue)
    )
    let count = 10

    self.backend.stubbedPostReceiptResult = .success(self.createCustomerInfo(nonSubscriptionProductID: nil))

    for _ in 0..<count {
        _ = try self.handleTransaction(transactionData)
    }

    expect(self.backend.invokedPostReceiptDataCount) == 1
    expect(self.mockTransaction.finishInvokedCount) == count

    expect(self.cache.postedTransactions) == [self.mockTransaction.transactionIdentifier]
}
```